python/ray/tests/test_nccl_channel.py

```diff
@@ -0,0 +1,260 @@
+# coding: utf-8
+import logging
+import os
+import sys
+import torch
+import mock
+import asyncio
+from collections import defaultdict
+
+import pytest
+
+import ray
+import ray.cluster_utils
+import ray.experimental.channel as ray_channel
+from ray.experimental.channel.torch_tensor_type import TorchTensorType
+from ray.experimental.channel.torch_tensor_nccl_channel import (
+    _init_nccl_group,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote(num_cpus=0)
+class Barrier:
+    """
+    Barrier that blocks the given number of actors until all actors have
+    reached the barrier. This is used to mock out blocking NCCL ops.
+    """
+
+    def __init__(self, num_actors=2):
+        self.num_actors = num_actors
```

Comment:
Throughout this file, should these (private) class members have an underscore prefix?

Reply:
It doesn't matter since these are just for testing.